### DISCUSSION TITLE
12 febbraio 2024 - esercizio 1

### DISCUSSION THREAD

**Discussion by bobbylinux** [sapienzastudentsnetwork/basi-di-dati-1#95] [Esami]
Si consideri il seguente schema di una base di dati di una casa discografica:

<div align="center">

MUSICISTA(<ins>ID</ins>, Nome, Cognome, DataNascita, DataMorte, Nazionalità, Strumento)

BAND(<ins>BandID</ins>, Nome, AnnoScioglimento)

MEMBRO(<ins>IDmusicista</ins>, <ins>IDband</ins>, AnnoInizio)

BRANO(<ins>ID</ins>, IDband, Titolo, Durata)

ALBUM(<ins>IDalbum</ins>, IDband, Titolo, Anno, Genere)

INCLUSO(<ins>IDbrano, IDalbum</ins>)

</div>

NOTE.
- Se un musicista è ancora vivo, allora DataMorte = 00/00/0000

- Se la band è ancora in attività, allora AnnoScioglimento = 0000

- Un musicista può suonare o aver suonato in più di una band.

- La durata dei brani è espressa in secondi.

1a) Trovare i dati di tutti i musicisti che hanno suonato in una band con almeno un Violoncello e che ha pubblicato un album del genere “ProgRock” dopo il 1975, in cui c’è un brano che dura almeno 12 minuti.

1b) Trovare i dati di tutti i musicisti che sono stati fondatori di almeno una band. Per fondatori, si intende che non ci sono altri musicisti che hanno iniziato a suonare in quella band prima di loro.

**Comment by bobbylinux** [comment]
La mia soluzione

![Screenshot from 2024-02-13 16-00-07](https://github.com/sapienzastudentsnetwork/bdi2324/assets/7393062/9a252647-08c4-44ab-bd1e-9b9dc9bfa806)

**Comment by Alessandro700** [comment]
Per la prima query tutto ok, penso.. ma puo darsi che mi sbaglio, nella seconda query ok con il fatto di fare il prodotto cartesiano di membro, e con la selezione con il minore, ma credo manchi la sottrazione del prodotto catesiano di membro con il membro originale e inserire anche nel theta join la condizione membro1.idband != membro.idband
metto qui la mia soluzione:
<img width="1721" height="655" alt="image" src="https://github.com/user-attachments/assets/b04d35bb-aa2f-4410-92bf-c129a1c553dc" />
